docs/plugins/transform-request.md

```diff
@@ -236,6 +235,109 @@ Set header from authentication context:
 }
 ```
 
+#### Set headers from dynamic scope:
```

Comment:
We should not refer to dynamic scopes directly.

This is a generic feature and should be described as such.
One of the examples in this section could describe the dynamic scope scenario.

Reply:
Maybe something mentionint `advanced` and `pattern matching`